tests/integration/suite/daprd/resiliency/apps/retrymatchinggrpc.go

```diff
@@ -0,0 +1,196 @@
+		counter := c.(*atomic.Int32)
+		counter.Add(1)
+
+		return nil, status.Error(codes.Code(responseStatusCode), "error for key: "+key)
+	}
+
+	resiliency := `
+apiVersion: dapr.io/v1alpha1
+kind: Resiliency
+metadata:
+  name: myresiliency
+spec:
+  policies:
+    retries:
+      DefaultAppRetryPolicy:
+        policy: constant
+        duration: 10ms
+        maxRetries: 3
+        matching:
+          gRPCStatusCodes: 1,2,5-7
+          httpStatusCodes: 500-599
+`
+	srv1 := app.New(t, app.WithOnInvokeFn(onInvoke))
+	srv2 := app.New(t, app.WithOnInvokeFn(onInvoke))
+
+	d.daprd1 = daprd.New(t,
+		daprd.WithAppProtocol("grpc"),
+		daprd.WithAppPort(srv1.Port(t)),
+		daprd.WithResourceFiles(resiliency),
+	)
+	d.daprd2 = daprd.New(t,
+		daprd.WithAppProtocol("grpc"),
+		daprd.WithAppPort(srv2.Port(t)),
+		daprd.WithResourceFiles(resiliency),
+	)
+
+	d.counters = sync.Map{}
+
+	return []framework.Option{
+		framework.WithProcesses(srv1, srv2, d.daprd1, d.daprd2),
+	}
+}
+
+func (d *retrymatchinggrpc) getCount(key string) int {
+	c, ok := d.counters.Load(key)
+	if !ok {
+		return 0
+	}
+
+	return int(c.(*atomic.Int32).Load())
+}
+
+func (d *retrymatchinggrpc) Run(t *testing.T, ctx context.Context) {
+	d.daprd1.WaitUntilRunning(t, ctx)
+	d.daprd2.WaitUntilRunning(t, ctx)
+
+	scenarios := []struct {
+		title              string
+		statusCode         int
+		expectedStatusCode int
+		expectRetries      bool
+	}{
+		{
+			title:              "success not in matching list 0",
+			statusCode:         0,
+			expectedStatusCode: 0,
+			expectRetries:      false,
+		},
+		{
+			title:              "error code not in matching list 14",
+			statusCode:         14,
+			expectedStatusCode: 14,
+			expectRetries:      false,
+		},
+		{
+			title:              "error code in matching list 1",
+			statusCode:         1,
+			expectedStatusCode: 1,
+			expectRetries:      true,
+		},
+		{
+			title:              "invalid status code 20",
+			statusCode:         20,
+			expectedStatusCode: 20,
+			expectRetries:      false,
+		},
+	}
+
+	//nolint:staticcheck
+	conn, err := grpc.DialContext(ctx, d.daprd1.GRPCAddress(),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithBlock(),
+	)
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, conn.Close()) })
+
+	for _, scenario := range scenarios {
+		t.Run(scenario.title, func(t *testing.T) {
+			key := uuid.NewString()
+			statusCodeStr := strconv.Itoa(scenario.statusCode)
+			ctxWithMetadata := metadata.AppendToOutgoingContext(
+				ctx, "dapr-app-id", d.daprd2.AppID())
+
+			// Uses AppCallback API since it is already present in the codebase.
+			// This will make the sidecar use the gRPC proxy way, it just happens
+			// to be the AppCallback API (could be MyMadeUpService) but it does not
+			// make a difference here.
+			_, err := rtv1.NewAppCallbackClient(conn).OnInvoke(ctxWithMetadata, &commonv1.InvokeRequest{
+				Method: key,
+				Data:   &anypb.Any{Value: []byte(statusCodeStr)},
```

Comment:
I understood this resiliency covers more than just invocation. Are we testing that? Also remember we have `Invoke`, as well as the gRPC "`proxy`" messaging for custom gRPC rpcs. I don't think the later is testing here.

https://github.com/dapr/dapr/blob/8b8b6052b49391985d9b95eec08f8729ed1562ab/tests/integration/suite/daprd/serviceinvocation/grpc/appapitoken/remotereceivernotoken.go#L54

Reply:
There is this `TestService` [here](https://github.com/dapr/dapr/tree/94fe7b977ff8b34fde5677af662bb0736807f74b/tests/integration/framework/process/grpc/app/proto) which is used by few tests. Registering the server in the test app [here](https://github.com/dapr/dapr/blob/9c69f64aa8718f4d7cfabdac0479f5416c96c05e/tests/integration/framework/process/grpc/app/app.go#L59) and use the Ping handler [here](https://github.com/dapr/dapr/blob/94fe7b977ff8b34fde5677af662bb0736807f74b/tests/integration/framework/process/grpc/app/options.go#L99)
I've added the gRPC proxy tests to run the same gRPC scenarios.